oeps/oep-0008.rst

```diff
@@ -0,0 +1,323 @@
+==================================
+OEP-8: Unit Testing Best Practices
+==================================
+
++---------------+-------------------------------------------+
+| OEP           | :doc:`OEP-8 </oeps/oep-0008>`             |
++===============+===========================================+
+| Title         | Unit Testing Best Practices               |
++---------------+-------------------------------------------+
+| Last Modified | 2016-09-09                                |
++---------------+-------------------------------------------+
+| Author        | Jesse Zoldak                              |
++---------------+-------------------------------------------+
+| Arbiter       |                                           |
++---------------+-------------------------------------------+
+| Status        | Draft                                     |
++---------------+-------------------------------------------+
+| Type          | Best Practice                             |
++---------------+-------------------------------------------+
+| Created       | 2016-09-09                                |
++---------------+-------------------------------------------+
+| `Resolution`  |                                           |
++---------------+-------------------------------------------+
+| `Replaces`    |                                           |
++---------------+-------------------------------------------+
+| `Replaced-By` |                                           |
++---------------+-------------------------------------------+
+
+Abstract
+========
+
+The edX ecosystem has grown from a handful of developers working on a few
+repos to many developers working on many repos. As such, questions arise as to
+tools, frameworks, and patterns to use in unit testing. This proposal is
+intended to codify best practices and to provide a reference for contributors
+when making choices.
```

Comment:
For example.  Say we have a repo, edx/ned-tools. It uses pytest for testing, so it is in conformance with OEP-8.  Then a new test runner (holotest) appears on the scene, and we want to try it out.  We switch over ned-tools to use holotest.  From an OEP-8 point of view, that repo has regressed: it was in compliance, and now it is not.  Is this OK?  It's too soon to say that holotest is the best practice.